### PR TITLE
Added fields in appData and sensorData

### PR DESCRIPTION
**Pull request by Nicolasdejean**
added optional waterLevel, GPS and flush.

### CODE REVIEW

File: src/api/appData.proto [hunk 1]
```diff
@@ -4,4 +4,5 @@ package api;
 
 message AppData {
         optional bool light = 1;
+        optional bool flush = 2;
```
Comment (johanstokking) [inline src/api/appData.proto]: Call it `activate`, that's more generic

File: src/api/sensorData.proto [hunk 2]
```diff
@@ -2,12 +2,21 @@ syntax = "proto2";
 
 package api;
 
+message GPS {
```
Comment (johanstokking) [inline src/api/sensorData.proto]: `Location`

GPS is a system

File: src/api/sensorData.proto [hunk 3]
```diff
@@ -2,12 +2,21 @@ syntax = "proto2";
...
+
 message SensorData {
         optional bool motion = 1;
         optional uint32 water = 2;
         optional float temperature_celcius = 3;
         optional float temperature_fahrenheit = 4;
         optional float humidity = 5;
 
+        optional GPS gps = 6;
+        optional uint32 waterLevel = 7;
```
Comment (johanstokking) [inline src/api/sensorData.proto]: Consistent naming; `water_level`

**Comment by FokkeZB** [comment]
@johanstokking, @Nicolasdejean addressed your requests for changes

**Comment by FokkeZB** [comment]
@johanstokking, @Nicolasdejean addressed your requests for changes.

**Comment by Nicolasdejean** [comment]
We can wait to merge this one, I'm going to add fields for downlink and uplink for waternet.

File: src/api/appData.proto [hunk 4]
```diff
@@ -4,4 +4,6 @@ package api;
 
 message AppData {
         optional bool light = 1;
-}
\ No newline at end of file
+        optional bool activate = 2;
+        optional float deactivation_limit = 3;
```
Comment (johanstokking) [inline src/api/appData.proto]: Insert an `activation_limit` too, as field 2

File: src/api/sensorData.proto [hunk 5]
```diff
@@ -2,12 +2,24 @@ syntax = "proto2";
...
         optional uint32 water = 2;
         optional float temperature_celcius = 3;
         optional float temperature_fahrenheit = 4;
         optional float humidity = 5;
 
+        optional Location location = 6;
+        optional float water_distance = 7;
+        optional bool not_raining = 8;
+        optional float pump_activated_timer = 9;
+        optional bool pump_activated = 10;
```
Comment (johanstokking) [inline src/api/sensorData.proto]: Let's make this more generic;

- `level`
- `triggered`
- `actuator_active`
- `actuator_time`

Note that the last two fields doesn't sound as if they have anything to do with `SensorData`. Maybe rename the whole thing to `DeviceData`?

File: docs/TheThingsMessage.md [hunk 6]
```diff
@@ -41,14 +41,14 @@ You can also add other analog readings.
...
+## Method: encodeDeviceData
 Encode the message you want to send.
 
 ```c
-static void encodeSensorData(sensordata_t *data, byte **buffer, size_t *size);
+static void encodeDeviceData(devicedata_t *data, byte **buffer, size_t *size);
 ```
 
-- `sensordata_t *data`: Structure containing all the message we can send.
+- `devicedata_t *data`: Structure containing all the message we can send.
```
Comment (johanstokking) [inline docs/TheThingsMessage.md]: Structure containing typical fields that devices send

File: examples/Message/Receive/Receive.ino [hunk 7]
```diff
@@ -33,24 +33,24 @@ void setup()
...
-  sensorData.has_motion = true;
-  sensorData.has_water = true;
-  sensorData.has_temperature_celcius = false;
-  sensorData.has_temperature_fahrenheit = false;
-  sensorData.has_humidity = false;
+  data.has_motion = true;
+  data.has_water = true;
+  data.has_temperature_celcius = false;
+  data.has_temperature_fahrenheit = false;
+  data.has_humidity = false;
```
Comment (johanstokking) [inline examples/Message/Receive/Receive.ino]: Default is false already, remove three lines here

**Pull request merged**